crypto/kyber/pqcrystals_kyber_ref_common/poly.c

```diff
@@ -164,16 +166,17 @@ void poly_frombytes(poly *r, const uint8_t a[KYBER_POLYBYTES])
 void poly_frommsg(poly *r, const uint8_t msg[KYBER_INDCPA_MSGBYTES])
 {
   unsigned int i,j;
-  int16_t mask;
+  crypto_word_t mask;
 
 #if (KYBER_INDCPA_MSGBYTES != KYBER_N/8)
 #error "KYBER_INDCPA_MSGBYTES must be equal to KYBER_N/8 bytes!"
 #endif
 
   for(i=0;i<KYBER_N/8;i++) {
     for(j=0;j<8;j++) {
-      mask = -(int16_t)((msg[i] >> j)&1);
-      r->coeffs[8*i+j] = mask & ((KYBER_Q+1)/2);
+      mask = constant_time_lt_w(0, (msg[i] >> j) & 1);
+      r->coeffs[8*i+j] = (int16_t) constant_time_select_w(mask,
```

Comment:
can we add a comment that the conversion to `int16_t` works only because the possible values (`(KYBER_Q+1)/2`, 0) are such that no error is caused by the cast? maybe add an assert as well?